### ISSUE TITLE
linkedlist lint should not warn when implementing a trait for LinkedList

### ISSUE THREAD

**Issue by fhartwig** [Manishearth/rust-clippy#605]
Running clippy on serde results in the following warning:

```
src/ser/impls.rs:291:23: 291:36 warning: I see you're using a LinkedList! Perhaps you meant some other data structure?, #[warn(linkedlist)] on by default
src/ser/impls.rs:291 impl<T> Serialize for LinkedList<T>
                                           ^~~~~~~~~~~~~
src/ser/impls.rs:291:23: 291:36 help: a VecDeque might work
for further information visit https://github.com/Manishearth/rust-clippy/wiki#linkedlist
```

Ignoring my general reservations against this lint, a trait impl on `LinkedList` isn't the same as using a linked list, so I think we should ignore cases like this.

**Comment by birkenfeld** [comment]
:+1:

**Comment by mcarton** [comment]
I already noticed that on *quickcheck* in #553. I’ll make a PR tomorrow.
